proto/substrait/algebra.proto

```diff
@@ -47,6 +47,12 @@ message RelCommon {
 
     substrait.extensions.AdvancedExtension advanced_extension = 10;
 
+    // Save or load a system-specific datastructure for use in optimizing a remote operation.
+    // The value corresponds to a plan unique number for that datastructure.  Any particular
+    // computation may be saved only once but it may be loaded multiple times.
+    optional int32 save_intermediate_computation = 11;
```

Comment:
I like the start. We probably need a richer interface though. At a minimum, the computation would likely have some metadata describing what they are (an Any here). If we're going to make a place for the general concept, I think we should make a place for that. Secondly, there are probably repeated. E.g. save the bloomfilter and the hashtable as two separate things to save. Or read the two hashtables from two other operators.

So maybe something like SaveComputation and LoadComputation messages, with each being repeated?

Reply:
I was considering repeated as a future possibility so I had used optional here.  I've switched it to repeated.

I've also added type although I avoided using the any type as that would make the optimization system specific.  By defining broad classes of types and using an instance number I'm hoping that the optimizations will be more portable.